src/components/GoogleAuthentication/GoogleAuthentication.js

```diff
@@ -0,0 +1,89 @@
+import React, { Component } from "react";
+import { gapi, loadAuth2 } from "gapi-script";
+
+//import UserCard from "./UserCard";
+import "./GoogleAuthentication.css";
+
+class GoogleAuthentication extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      user: null
+    };
+  }
+  async componentDidMount() {
+    let auth2 = await loadAuth2(
+      "399957999889-j4a1hhcp9kdtk2cl5qkgrfv8cgme9fpe.apps.googleusercontent.com",
+      ""
+    );
+    if (auth2.isSignedIn.get()) {
+      this.updateUser(auth2.currentUser.get());
+    } else {
+      this.attachSignin(document.getElementById("customBtn"), auth2);
+    }
+  }
+
+  async componentDidUpdate() {
+    if (!this.state.user) {
+      let auth2 = await loadAuth2(
+        "399957999889-j4a1hhcp9kdtk2cl5qkgrfv8cgme9fpe.apps.googleusercontent.com",
+        "https://www.googleapis.com/auth/calendar"
+      );
+      this.attachSignin(document.getElementById("customBtn"), auth2);
+    }
+  }
+
+  updateUser(currentUser) {
+    let name = currentUser.getBasicProfile().getName();
+    let profileImg = currentUser.getBasicProfile().getImageUrl();
+    this.setState({
+      user: {
+        name: name,
+        profileImg: profileImg
+      }
+    });
+  }
+
+  attachSignin(element, auth2) {
+    auth2.attachClickHandler(
+      element,
+      {},
+      googleUser => {
+        this.updateUser(googleUser);
+      },
+      error => {
+        console.log(JSON.stringify(error));
+      }
+    );
+  }
+
+  signOut = () => {
```

Comment:
We can add this sign out method to the SidebarNav component in the onClick prop on the Button component in SidebarNav. We can rename it to googleSignOut and can call it below `Auth.signOut(..)` but above `CometChat.logout`. We might need to create a new React context object for GoogleUserContext, similar to the src/UserContext.js file. That will let you use the useContext hook in SidebarNav to get access to the GoogleUserContext object we should set after the user signs in to our app.